packages/start/test/start.test.js

```diff
@@ -0,0 +1,214 @@
+
+    t.after(async () => {
+      await server.close()
+    })
+
+    const address = await server.start()
+    // The address should be a valid URL.
+    new URL(address) // eslint-disable-line no-new
+  })
+
+  await t.test('can build a db server', async (t) => {
+    const configFile = join(fixturesDir, 'dbApp', 'platformatic.db.json')
+    const config = await loadConfig({}, ['-c', configFile])
+    const server = await buildServer(config.configManager.current)
+
+    t.after(async () => {
+      await server.close()
+    })
+
+    const address = await server.start()
+    // The address should be a valid URL.
+    new URL(address) // eslint-disable-line no-new
+  })
+})
+
+test('start()', async (t) => {
+  await t.test('can start a service server', (t) => {
+    return new Promise((resolve, reject) => {
+      const scriptFile = join(fixturesDir, 'starter.js')
+      const configFile = join(fixturesDir, 'serviceApp', 'platformatic.service.json')
+      const child = spawn(process.execPath, [scriptFile, configFile])
+
+      child.on('error', reject)
+      child.on('exit', (exitCode) => {
+        if (exitCode === 42) {
+          resolve()
+        } else {
+          reject(new Error(`unexpected exit code: ${exitCode}`))
+        }
+      })
+    })
+  })
+
+  await t.test('can start a db server', (t) => {
+    return new Promise((resolve, reject) => {
+      const scriptFile = join(fixturesDir, 'starter.js')
+      const configFile = join(fixturesDir, 'dbApp', 'platformatic.db.json')
+      const child = spawn(process.execPath, [scriptFile, configFile])
+
+      child.on('error', reject)
+      child.on('exit', (exitCode) => {
+        if (exitCode === 42) {
+          resolve()
+        } else {
+          reject(new Error(`unexpected exit code: ${exitCode}`))
+        }
+      })
+    })
+  })
+})
+
+test('startCommand()', async (t) => {
+  await t.test('can start a server', (t) => {
+    return new Promise((resolve, reject) => {
+      const scriptFile = join(fixturesDir, 'start-command.js')
+      const configFile = join(fixturesDir, 'serviceApp', 'platformatic.service.json')
+      const child = spawn(process.execPath, [scriptFile, configFile])
+
+      child.on('error', reject)
+      child.on('exit', (exitCode) => {
+        if (exitCode === 42) {
+          resolve()
+        } else {
+          reject(new Error(`unexpected exit code: ${exitCode}`))
+        }
+      })
+    })
+  })
+
+  await t.test('exits on error', (t) => {
+    return new Promise((resolve, reject) => {
+      const scriptFile = join(fixturesDir, 'start-command.js')
+      const configFile = join(fixturesDir, 'serviceApp', 'platformatic.not-found.json')
+      const child = spawn(process.execPath, [scriptFile, configFile])
+
+      child.on('error', reject)
+      child.on('exit', (exitCode) => {
+        if (exitCode === 1) {
+          resolve()
+        } else {
+          reject(new Error(`unexpected exit code: ${exitCode}`))
+        }
+      })
```

Comment:
you can use https://nodejs.org/api/events.html#eventsonceemitter-name-options instead of `new Promise()`